### PR TITLE
Changes for adding the infrastructure for optimizer

### PR DESCRIPTION
**Pull request by sulabh-msft**
## Purpose

* ...
Added code for creating the Azure load testing optimizer infrastructure
- Azure load Testing Resource
- Optimizer  - Test Profile
- Optimizer Test Profile Run

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: infra/abbreviations.json [hunk 1]
```diff
@@ -55,6 +55,9 @@
     "kubernetesConnectedClusters": "arck",
     "kustoClusters": "dec",
     "kustoClustersDatabases": "dedb",
+    "loadtesting": "alt-",
+    "loadtestingTests": "altt-",
+    "loadtestingProfiles": "altp-",
```
Comment (mitsha-microsoft) [inline infra/abbreviations.json]: nit: `alttp`

File: infra/core/loadtesting/loadtests.bicep [hunk 2]
```diff
@@ -0,0 +1,18 @@
+param name string
+param location string = resourceGroup().location
+param tags object = {}
+
+resource loadtests 'Microsoft.LoadTestService/loadTests@2022-12-01' = {
+  name: name
+  location: location
+  tags: tags
+  identity: {
+    type: 'SystemAssigned'
```
Comment (mitsha-microsoft) [inline infra/core/loadtesting/loadtests.bicep]: Do we need MSI for ALT?

File: scripts/postdeploy.ps1 [hunk 3]
```diff
@@ -7,4 +7,232 @@ foreach ($tool in $tools) {
...
+function Log($String) {
+    Write-Debug $String
+}
+
+# Ensure az load extension is installed
+az extension add --name load
+
+# Get Dataplane URL for Load Testing Resource
+$DataPlaneURL = az load show -n $LoadTestResourceName -g $ResourceGroupName --query dataPlaneURI
+$DataPlaneURL = "https://" + $DataPlaneURL.Trim('"')
```
Comment (mitsha-microsoft) [inline scripts/postdeploy.ps1]: We can remove this since its part of the template output? `LOADTEST_DP_URL` ?

File: scripts/postdeploy.sh [hunk 4]
```diff
@@ -7,4 +7,220 @@ for cmd in "${commands[@]}"; do
...
+# Create Test Profile Run
+TestProfileRunRequest=$(cat <<EOF
+{
+  "testProfileId": "$TestProfileId",
+  "displayName": "$TestProfileRunDisplayName"
+}
+EOF
+)
+TestProfileRunId=$(uuidgen)
+TestProfileRunURL="$DataPlaneURL/test-profile-runs/$TestProfileRunId?api-version=$ApiVersion"
```
Comment (mitsha-microsoft) [inline scripts/postdeploy.sh]: Script seems to have an issue, it didnt call Patch test profile run?

File: infra/core/loadtesting/loadtests.bicep (hunk 2 above)
Comment (sulabh-msft) [reply infra/core/loadtesting/loadtests.bicep]: Removed

File: scripts/postdeploy.sh (hunk 4 above)
Comment (sulabh-msft) [reply scripts/postdeploy.sh]: WIP.. Coming in next iteration

**Review by nzthiago — APPROVED** [review]

File: scripts/postdeploy.ps1 [hunk 5]
```diff
@@ -7,4 +7,243 @@ foreach ($tool in $tools) {
...
+# Function App Details
+$FunctionAppName = ${env:AZURE_FUNCTION_NAME}
+$FunctionAppTriggerName = ${env:AZURE_FUNCTION_APP_TRIGGER_NAME}
+$FunctionAppResourceId = ${env:AZURE_FUNCTION_APP_RESOURCE_ID}
+
+# ALT Resource Details
+$LoadTestResourceName = ${env:AZURE_LOADTEST_RESOURCE_NAME}
+$ResourceGroupName = ${env:RESOURCE_GROUP}
+$TestId = ${env:LOADTEST_TEST_ID}
+$DataPlaneURL = "https://" + ${LOADTEST_DP_URL}.Trim('"')
```
Comment (mitsha-microsoft) [inline scripts/postdeploy.ps1]: Change to `${env:LOADTEST_DP_URL}` ?

File: scripts/postdeploy.sh (hunk 4 above)
Comment (sulabh-msft) [reply scripts/postdeploy.sh]: Added it in this iteration

**Pull request merged**